docarray/base_doc/doc.py

```diff
@@ -73,6 +74,43 @@ class Config:
         validate_assignment = True
         _load_extra_fields_from_protobuf = False
 
+    @classmethod
+    def _add_fields(cls, **field_definitions):
```

Comment:
```suggestion
    def _add_fields(cls, **field_definitions) -> None:
```

Reply:
I guess this is used for add the root it and removing it right ?

I am not a fan of this method because it change the original class.

We should rather create a new class that inherit from this class and add the root_id/parent_id information